metricbeat/module/logstash/node_stats/data.go

```diff
@@ -174,9 +176,20 @@ func eventMapping(r mb.ReporterV2, content []byte, isXpack bool) error {
 	}
 
 	var pipelines []PipelineStats
+
+	// The version from which we expect the node stats API to return a hash with pipeline documents. This is really just a formality so that
+	// unit tests with much older fixture versions still work.
+	var PipelineDocumentsContainHashVersion = version.MustNew("7.3.0")
+	var StatsVersion = version.MustNew(nodeStats.Version)
+	pipelineDocumentsShouldContainHash := !StatsVersion.LessThan(PipelineDocumentsContainHashVersion)
+
 	for pipelineID, pipeline := range nodeStats.Pipelines {
-		pipeline.ID = pipelineID
-		pipelines = append(pipelines, pipeline)
+		if pipelineDocumentsShouldContainHash && (pipeline.Hash == "" || pipeline.Vertices == nil) {
+			logger.Warn("Pipeline document was discarded due to missing properties. This can happen when the Logstash node stats API is polled before the pipeline setup has completed.")
```

Comment:
Would it be useful to include the pipelineID here ?

Reply:
It would, good point.